File: src/nl/rubensten/texifyidea/lang/OptionalArgument.java
package nl.rubensten.texifyidea.lang;

/**
 * @author Sten Wessel
 */
public class OptionalArgument extends Argument {

    OptionalArgument(String name) {
        this(name, Type.NORMAL);
    }

    OptionalArgument(String name, Type type) {
        super(name, type);
    }

    @Override
    public String toString() {
        return "[" + getName() + "]";
    }
}

File: src/nl/rubensten/texifyidea/lang/OptionalArgument.kt
package nl.rubensten.texifyidea.lang

/**
 * @author Sten Wessel
 */
@Suppress("ConvertSecondaryConstructorToPrimary")
class OptionalArgument : Argument {

    @JvmOverloads
    internal constructor(name: String, type: Argument.Type = Argument.Type.NORMAL) : super(name, type)

    override fun toString() = "[$name]"
}